src/toast/templates/offset.py

```diff
@@ -511,11 +511,15 @@ def _project_signal(self, detector, amplitudes):
             step_length = self._step_length(
                 self.step_time.to_value(u.second), self._obs_rate[iob]
             )
-            for ivw, vw in enumerate(ob.view[self.view].detdata[self.det_data]):
-                n_amp_view = self._obs_views[iob][ivw]
+            for iview, (todview, flagview) in enumerate(zip(
+                    ob.view[self.view].detdata[self.det_data],
+                    ob.view[self.view].detdata[self.det_flags],
+            )):
+                n_amp_view = self._obs_views[iob][iview]
+                mask = (flagview[detector] & self.det_flag_mask) == 0
```

Comment:
Thanks for this fix.  I'll have to merge this to the branch with GPU testing, since this code has changed in that branch.